rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -1842,39 +1844,74 @@ int LuaSyncedCtrl::DestroyUnit(lua_State* L)
  * @function Spring.TransferUnit
  * @param unitID integer
  * @param newTeamID integer
- * @param given boolean? (Default: `true`) if false, the unit is captured.
+ * @param given boolean (Default: `true`) if false, the unit is captured.
  * @return nil
```

Comment:
Ok, similar to AllowUnitTransfer, I'm realizing we don't actually need TransferUnitWithReason. We can just call TransferUnit, deprecate the third parameter, and ignore the third parameter altogether.